state-sync/state-sync-v2/data-streaming-service/src/streaming_client.rs

```diff
@@ -68,23 +68,30 @@ pub trait DataStreamingClient {
     ) -> Result<DataStreamListener, Error>;
 
     /// Continuously streams transaction outputs with proofs as the blockchain
-    /// grows. The stream starts at `start_version` and `start_epoch` (inclusive).
+    /// grows. The stream starts at `known_version + 1` (inclusive) and
+    /// `known_epoch`, where the `known_epoch` is expected to be the epoch
+    /// that contains `known_version + 1`, i.e., any epoch change at
+    /// `known_version` must be noted by the client.
```

Comment:
This looks confusing to me - can you clarify why the client need to specify the `epoch` at all? `version` being a unique timestamp, that should be sufficient to identify the start of the stream?

Reply:
This is more of an implementation detail than an architectural one. The data streaming service needs to track epoch changes, so it needs to know what epoch it starts at. We make it the client responsibility to provide this because the streaming service doesn't have access to a database, so it can't simply look up the epoch for the given version. One could even argue that a streaming service like this probably shouldn't rely on a database at all (given that it's just a glorified pre-fetcher) and it might make sense to leverage this in other areas (where there is no direct database), e.g., a client :)